cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForVscConverters.java

```diff
@@ -0,0 +1,117 @@
+        this.parent = parent;
+        this.context = context;
+        mapping = new HashMap<>();
+    }
+
+    public static void initialize(VscConverterStationAdder adder) {
+        adder.setVoltageRegulatorOn(false)
+            .setReactivePowerSetpoint(0.0);
+    }
+
+    public void add(String vscConverterId, PropertyBag sm) {
+        if (mapping.containsKey(vscConverterId)) {
+            throw new CgmesModelException("VscConverter already added, IIDM VscConverter Id: " + vscConverterId);
+        }
+
+        CgmesRegulatingControlForVscConverter rd = new CgmesRegulatingControlForVscConverter();
+        rd.vscRegulation = sm.getLocal("qPccControl");
+        rd.voltageSetpoint = sm.asDouble("targetUpcc");
+        rd.reactivePowerSetpoint = -sm.asDouble("targetQpcc");
+        rd.pccTerminal = sm.getId("PccTerminal");
+        mapping.put(vscConverterId, rd);
+    }
+
+    private static VscRegulation decodeVscRegulation(String qPccControl) {
+        if (qPccControl != null) {
+            if (qPccControl.endsWith("voltagePcc")) {
+                return VscRegulation.VOLTAGE;
+            } else if (qPccControl.endsWith("reactivePcc")) {
+                return VscRegulation.REACTIVE_POWER;
+            }
+        }
+        return null;
+    }
+
+    void applyRegulatingControls(Network network) {
+        network.getVscConverterStationStream().forEach(this::apply);
+    }
+
+    private void apply(VscConverterStation vscConverter) {
+        CgmesRegulatingControlForVscConverter rd = mapping.get(vscConverter.getId());
+        apply(vscConverter, rd);
+    }
+
+    private void apply(VscConverterStation vscConverter, CgmesRegulatingControlForVscConverter rc) {
+        if (rc == null) {
+            return;
+        }
+
+        VscRegulation vscRegulation = decodeVscRegulation(rc.vscRegulation);
+        if (vscRegulation == VscRegulation.VOLTAGE) {
+            setRegulatingControlVoltage(rc, vscConverter);
+        } else if (vscRegulation == VscRegulation.REACTIVE_POWER) {
+            setRegulatingControlReactivePower(rc, vscConverter);
+        } else {
+            context.ignored(rc.vscRegulation, "Unsupported regulation mode for vscConverter " + vscConverter.getId());
+        }
+    }
+
+    private void setRegulatingControlVoltage(CgmesRegulatingControlForVscConverter rc, VscConverterStation vscConverter) {
+
+        vscConverter
+            .setVoltageSetpoint(rc.voltageSetpoint)
+            .setReactivePowerSetpoint(0.0)
+            .setRegulatingTerminal(parent.findRegulatingTerminal(rc.pccTerminal, false))
+            .setVoltageRegulatorOn(true);
+    }
+
+    private void setRegulatingControlReactivePower(CgmesRegulatingControlForVscConverter rc, VscConverterStation vscConverter) {
+
+        vscConverter
+            .setVoltageSetpoint(0.0)
+            .setReactivePowerSetpoint(rc.reactivePowerSetpoint)
+            .setRegulatingTerminal(parent.findRegulatingTerminal(rc.pccTerminal, false))
```

Comment:
Indeed the remote regulating terminal only works for voltage regulation. For remote reactive regulation, you have to use the dedicated extension. 
I have a question : finally, where is our precedent fix that fix the remote regulation of VSC coming from CGMES ? It was a conversion from remote to local. Do you remember?